### PR TITLE
Task2_bash_script

### PR DESCRIPTION
**Pull request by IhorRyzhov**
Implementation of bash script

@marynasergiyenko @a-vodka @AleksandrBulyshchenko @galkinletter please look at these commits

### CODE REVIEW

File: 01-Git_Bash_Make/Task2_bash_script/script [hunk 1]
```diff
@@ -0,0 +1 @@
+#!/bin/bash
```
Comment (AleksandrBulyshchenko) [inline 01-Git_Bash_Make/Task2_bash_script/script]: While it isn't strictly required, it's preferable to add `.sh` extension to shell scripts.

Comment (AleksandrBulyshchenko) [reply 01-Git_Bash_Make/Task2_bash_script/script]: Here and later commit messages seem to be waiting for some cleanup.

File: 01-Git_Bash_Make/Task2_bash_script/typescript [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
[22d[K[3;16H[?12l[?25h[?25l:[?12l[?25h[?25l [?12l[?25h[?25lA[?12l[?25h[?25ld[?12l[?25h[?25ld[?12l[?25h[?25l [?12l[?25h[?25le[?12l[?25h[?25lm[?12l[?25h[?25lp[?12l[?25h[?25lt[?12l[?25h[?25ly[?12l[?25h[?25l [?12l[?25h[?25ls[?12l[?25h[?25lc[?12l[?25h[?25lr[?12l[?25h[?25li[?12l[?25h[?25lp[?12l[?25h[?25lt[?12l[?25h[?25l [?12l[?25h[?25lf[?12l[?25h[?25li[?12l[?25h[?25ll[?12l[?25h[?25le[?12l[?25h[?25l
[22d(B[0;7mСохранить изменённый буфер?  (При ответе "Нет" все изменения будут ПОТЕРЯНЫ.)   [23;1H Y(B[m Да[K
[24d(B[0;7m N(B[m Нет          (B[0;7m^C(B[m Отмена[K[22;79H[?12l[?25h[?25l
[J[24;80H[?12l[?25h[24;1H[?1049l[23;0;0t
[?1l>
[KОтмена коммита из-за пустого сообщения коммита.
+]0;igor@igor-Vostr
```
Comment (AleksandrBulyshchenko) [inline 01-Git_Bash_Make/Task2_bash_script/typescript]: What is this file for?

File: 01-Git_Bash_Make/Task2_bash_script/script [hunk 3]
```diff
@@ -1 +1,6 @@
 #!/bin/bash
+
+directory=/tmp/guesanumber
```
Comment (AleksandrBulyshchenko) [inline 01-Git_Bash_Make/Task2_bash_script/script]: Prefer capitalised names for variables (at least global).

File: 01-Git_Bash_Make/Task2_bash_script/script [hunk 4]
```diff
@@ -1 +1,6 @@
...
+
+cp -R /home/igor/gl-kernel-basecamp-2019/01-Git_Bash_Make/Task1_game $directory
```
Comment (AleksandrBulyshchenko) [inline 01-Git_Bash_Make/Task2_bash_script/script]: Prefer to use braces syntax for variable expansion in scripts and quotes for whitespace sensitive strings.
like `"${VAR}"`

File: 01-Git_Bash_Make/Task2_bash_script/script [hunk 5]
```diff
@@ -2,5 +2,12 @@
 
 directory=/tmp/guesanumber
 
-cp -R /home/igor/gl-kernel-basecamp-2019/01-Git_Bash_Make/Task1_game $directory
+if ! [ -d $directory ]
```
Comment (AleksandrBulyshchenko) [inline 01-Git_Bash_Make/Task2_bash_script/script]: IMO it's preferable to keep all the condition expession (including inversion) within test statement.
And as I've already said - keep string arguments in quotes (`"${VAR}"`).

File: 01-Git_Bash_Make/Task2_bash_script/script (hunk 4 above)
Comment (AleksandrBulyshchenko) [reply 01-Git_Bash_Make/Task2_bash_script/script]: You have hard coded absolute path for the project within repository.
This is definitely looks wrong.
While it works for your current directories, you're commiting it to the repository.
What assumes you expect this repository to be fetched by others to any location where your script won't work due to the hardcoded path.

File: 01-Git_Bash_Make/Task2_bash_script/script [hunk 6]
```diff
@@ -5,6 +5,7 @@ directory=/tmp/guesanumber
 if ! [ -d $directory ]
 then
 	cp -R /home/igor/gl-kernel-basecamp-2019/01-Git_Bash_Make/Task1_game $directory
+	tar -cvzf $directory/Task1_game.tar.gz $directory
```
Comment (AleksandrBulyshchenko) [inline 01-Git_Bash_Make/Task2_bash_script/script]: While not strictly prohibited, placing the tarball in the directory being archived is bad idea - the procedure loses reenterability.

File: 01-Git_Bash_Make/Task2_bash_script/script [hunk 7]
```diff
@@ -6,6 +6,8 @@ if ! [ -d $directory ]
 then
 	cp -R /home/igor/gl-kernel-basecamp-2019/01-Git_Bash_Make/Task1_game $directory
 	tar -cvzf $directory/Task1_game.tar.gz $directory
+	mkdir /home/igor/gl-kernel-basecamp-2019/01-Git_Bash_Make/Task2_bash_script/release
+	cp $directory/Task1_game.tar.gz /home/igor/gl-kernel-basecamp-2019/01-Git_Bash_Make/Task2_bash_script/release
```
Comment (AleksandrBulyshchenko) [inline 01-Git_Bash_Make/Task2_bash_script/script]: You're using the same long path several times (I'm leaving aside hardcodding such paths) - make sense to define a variable.

Comment (a-vodka) [reply 01-Git_Bash_Make/Task2_bash_script/script]: It will be grate idea to define variable for temp and release dir

**Pull request closed without merging**